docs/apis/role.yaml

```diff
@@ -0,0 +1,816 @@
+openapi: 3.0.3
+info:
+  title: Role Management API
+  version: "1.0"
+  description: This API is used to manage roles with permission-based access control.
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+
+servers:
+  - url: https://{host}:{port}
+    variables:
+      host:
+        default: "localhost"
+      port:
+        default: "8090"
+
+tags:
+  - name: roles
+    description: Operations related to role management
+
+paths:
+  /roles:
+    get:
+      tags:
+        - roles
+      summary: List roles
+      parameters:
+        - $ref: '#/components/parameters/limitQueryParam'
+        - $ref: '#/components/parameters/offsetQueryParam'
+      responses:
+        "200":
+          description: List of roles
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/RoleListResponse'
+              example:
+                totalResults: 5
+                startIndex: 1
+                count: 2
+                roles:
+                  - id: "3fa85f64-5717-4562-b3fc-2c963f66afa6"
+                    name: "front-desk-agent"
+                    description: "Front desk agent role with booking and customer management permissions"
+                    organizationUnitId: "a839f4bd-39dc-4eaa-b5cc-210d8ecaee87"
+                  - id: "257e528f-eb24-48b6-884d-20460e190957"
+                    name: "hotel-manager"
+                    description: "Hotel manager with full property access"
+                    organizationUnitId: "a839f4bd-39dc-4eaa-b5cc-210d8ecaee87"
+                links:
+                  - href: "roles?offset=2&limit=2"
+                    rel: "next"
+        "400":
+          description: Bad request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              examples:
+                invalid-request-format:
+                  summary: Invalid request format
+                  value:
+                    code: "ROL-1001"
+                    message: "Invalid request format"
+                    description: "The request body is malformed or contains invalid data"
+                invalid-limit:
+                  summary: Invalid limit parameter
+                  value:
+                    code: "ROL-1011"
+                    message: "Invalid limit parameter"
+                    description: "The limit parameter must be a positive integer"
+                invalid-offset:
+                  summary: Invalid offset parameter
+                  value:
+                    code: "ROL-1012"
+                    message: "Invalid offset parameter"
+                    description: "The offset parameter must be a non-negative integer"
+        "500":
+          description: Internal server error
+          content:
+            text/plain:
```

Comment:
Shall we have json response for 500 internal server errors?

@darshanasbg are we planning to migrate to `plain/text` for server errors?